code/modules/projectiles/projectile/magic.dm

```diff
@@ -158,7 +158,8 @@
 						if("drone")
 							new_mob = new /mob/living/simple_animal/drone(M.loc)
 							var/mob/living/simple_animal/drone/D = new_mob
-							D.update_drone_hack()
+							spawn(50) //this should give us time for the rest of the proc to finish so a mob will be in the drone
```

Comment:
can you not just move the call to update_drone_Hack to after the point the mob is put in? because this is gross as fuck